Sally7/RequestExecutor/ConcurrentRequestExecutor.cs

```diff
@@ -92,6 +92,14 @@ public async ValueTask<Memory<byte>> PerformRequest(ReadOnlyMemory<byte> request
                     _ = await sendSignal.WaitAsync(cancellationToken).ConfigureAwait(false);
                     try
                     {
+                        // If we bail while sending the PLC might still respond to the data that was sent.
+                        // This both breaks the send-one-receive-one flow as well as it might end up
+                        // completing a new job that reused the ID.
+#if NET5_0_OR_GREATER || NETSTANDARD2_1_OR_GREATER
+                        await
+#endif
+                        using var closeOnCancel = cancellationToken.MaybeUnsafeRegister(socket.Close);
```

Comment:
As we use `try-finally` already I'd avoid the `using`-pattern here, and create that registration before the `try` and dispose it in the `finally`. Just to avoid the nested try-finally that is generated by the compiler.

Reply:
I always doubt whether I may assume some calls won't throw, because then it's safe to combine the try/finally blocks. Applies here as well, would love to hear your view on this.

Reply:
Here this is basically the first statement in the `try-catch`, so that one can be collapsed --> do the dispose manually in the finally.

Reply:
Changed this per recommendation. Considering all the safety measures in place for things that shouldn't happen (like checking if the memory is array based, which could theoretically pose an issue if someone would inject a different memory pool) it does feel like cutting corners, because an exception during `Dispose` would never release the `sendSignal`. I guess we can safely assume that never happens though, still somewhat bothers me 😉